### PR TITLE
INFO Log with header information

### PR DESCRIPTION
**Pull request by tjmoore4**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?
   - [ ] Have you added automated tests?

**Type of Changes:**

 
 - [x] New feature
 - [ ] Bug fix
 - [ ] Documentation
 - [ ] Testing enhancement
 - [ ] Other

**What is the new behavior (if this is a feature change)?**
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

Add header information as to existing logs, updated to INFO level
from DEBUG, and update related Go tests to match changes.

**Other Information**:
Issue: [sc-13581]

### CODE REVIEW

File: internal/upgradecheck/http.go [hunk 1]
```diff
@@ -136,8 +136,12 @@ func checkForUpgrades(ctx context.Context, versionString string, backoff wait.Ba
 		err = fmt.Errorf("received StatusCode %d", status)
 	}
 
+	if len(req.Header[clientHeader]) > 0 {
+		header = req.Header[clientHeader][0]
+	}
```
Comment (cbandy) [inline internal/upgradecheck/http.go:141]: ⛏️ `http.Header` has methods for this kind of thing.

```suggestion
	header = req.Header.Get(clientHeader)
```

File: internal/upgradecheck/http_test.go [hunk 2]
```diff
@@ -67,6 +68,14 @@ func TestCheckForUpgrades(t *testing.T) {
 	ctx := logging.NewContext(context.Background(), logging.Discard())
 	cfg := &rest.Config{}
 
+	checkData := func(header string) {
+		data := clientUpgradeData{}
+		err := json.Unmarshal([]byte(header), &data)
+		assert.NilError(t, err)
```
Comment (cbandy) [inline internal/upgradecheck/http_test.go:74]: 🔧 `checkData` is called from within subtests, so it should take a `*testing.T` for the asserts here.

🤔 I forget exactly what happens when asserting on the wrong `T`, but it makes it hard to diagnose the error. (It panics maybe?)

File: internal/upgradecheck/http.go (hunk 1 above)
Comment (tjmoore4) [reply internal/upgradecheck/http.go:141]: :+1: updated

File: internal/upgradecheck/http_test.go (hunk 2 above)
Comment (tjmoore4) [reply internal/upgradecheck/http_test.go:74]: Good note, I've updated the test accordingly and put a link to [the go docs](https://pkg.go.dev/github.com/stretchr/testify/assert#hdr-Assertions) that talk a bit about why `T` is passed.

**Review by benjaminjb — APPROVED** [review]
LGTM. Just a question (and a scope-creep question at that) but are there any other changes we should make to logging in this package?

File: internal/upgradecheck/http_test.go (hunk 2 above)
Comment (cbandy) [reply internal/upgradecheck/http_test.go:77]: `assert` here is [gotest.tools/v3/assert](https://pkg.go.dev/gotest.tools/v3/assert) so we should probably link to that, if it explains.

When it's the wrong `T`, it prints `test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test` with none of the messages from the assert package.

https://go.dev/play/p/gIVKDrd9rTR

Comment (tjmoore4) [reply internal/upgradecheck/http_test.go:77]: :thinking: Looking through that link and the code a bit, I can't find anything that exactly points that out. I'll update the note and remove the link.

**Comment by tjmoore4** [comment]
> LGTM. Just a question (and a scope-creep question at that) but are there any other changes we should make to logging in this package?

Nothing that stood out to me at this point. Was there anything in particular you were thinking?

**Review by cbandy — APPROVED** [review]

**Review by andrewlecuyer — APPROVED** [review]

**Pull request merged**